Sources_v3/StreamChatUI/ChatChannel/MessageComposerInputAccessoryViewController.swift

```diff
@@ -2,35 +2,161 @@
+        picker.sourceType = .photoLibrary
+        picker.delegate = self
+        return picker
+    }()
+    
+    // MARK: Setup
+    
+    override open func viewDidLoad() {
         super.viewDidLoad()
+        
+        setupInputView()
+        setUpLayout()
+    }
+    
+    func setupInputView() {
         inputView = composerView
-
-        guard let inputView = inputView else { return }
+        
         composerView.messageInputView.textView.delegate = self
-
+        composerView.attachmentButton.addTarget(self, action: #selector(showImagePicker), for: .touchUpInside)
+        composerView.sendButton.addTarget(self, action: #selector(sendMessage), for: .touchUpInside)
+        composerView.messageInputView.rightAccessoryButton.addTarget(
+            self,
+            action: #selector(dismissSlashCommand),
+            for: .touchUpInside
+        )
+        
+        composerView.attachmentsView.didTapRemoveItemButton = { [weak self] index in self?.imageAttachments.remove(at: index) }
+    }
+    
+    public func setUpLayout() {
+        guard let inputView = inputView else { return }
+        
         composerView.leadingAnchor.constraint(equalTo: view.leadingAnchor).isActive = true
         composerView.trailingAnchor.constraint(equalTo: view.trailingAnchor).isActive = true
         composerView.bottomAnchor.constraint(equalTo: view.bottomAnchor).isActive = true
         composerView.topAnchor.constraint(equalTo: inputView.topAnchor).isActive = true
     }
-}
-
-// MARK: - UITextViewDelegate
-
-extension MessageComposerInputAccessoryViewController: UITextViewDelegate {
-    func textViewShouldBeginEditing(_ textView: UITextView) -> Bool {
+    
+    // MARK: Button actions
+    
+    @objc func sendMessage() {
+        guard let text = composerView.messageInputView.textView.text,
+            !text.replacingOccurrences(of: " ", with: "").isEmpty
+        else { return }
+        
+        controller?.createNewMessage(text: text)
+        composerView.messageInputView.textView.text = ""
+    }
+    
+    @objc func showImagePicker() {
+        present(imagePicker, animated: true, completion: nil)
+    }
+    
+    @objc func dismissSlashCommand() {
+        composerView.messageInputView.rightAccessoryButton.isHidden = true
+        composerView.messageInputView.container.leftStackView.isHidden = true
+    }
+    
+    // MARK: Suggestions
+    
+    func showSuggestionsViewController() {
+        addChild(suggestionsViewController)
+        view.addSubview(suggestionsViewController.view)
+        suggestionsViewController.didMove(toParent: self)
+    }
+    
+    func dismissSuggestionsViewController() {
+        suggestionsViewController.removeFromParent()
+        suggestionsViewController.view.removeFromSuperview()
+    }
+    
+    // MARK: Attachments
+    
+    var imageAttachments: [UIImage] = [] {
+        didSet {
+            didUpdateImageAttachments()
+        }
+    }
```

Comment:
Future nit: For sending attachments we will also need to know the local URL of the attachment so we can upload it.